addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/SVN_IPG_LEDM_119.java

```diff
@@ -0,0 +1,253 @@
+                break;
+        }
+    }
+
+    @Override
+    public void CheckUsage() {
+        WebRequester status = new WebRequester(EWS_ENDPOINT_USAGE, config);
+        logger.debug("Checking {} usage", thing.getUID().toString());
+        WebRequesterChecker result = status.Do();
+        switch (result.Status()) {
+            case WebRequester.REQ_STATUS_OK:
+
+                // Get Ink Levels
+                NodeList consumableInk = result.Document().getDocumentElement()
+                        .getElementsByTagName("pudyn:Consumable");
+
+                try {
+                    for (int i = 0; i < consumableInk.getLength(); i++) {
+                        Element currInk = (Element) consumableInk.item(i);
+
+                        String inkName = currInk.getElementsByTagName("dd:MarkerColor").item(0).getTextContent();
+
+                        String consumeType = currInk.getElementsByTagName("dd:ConsumableTypeEnum").item(0)
+                                .getTextContent();
+
+                        if (consumeType.equalsIgnoreCase("printhead")) {
+                            continue;
+                        }
+
+                        String inkRemaining = currInk.getElementsByTagName("dd:ConsumableRawPercentageLevelRemaining")
+                                .item(0).getTextContent();
+
+                        logger.debug("{} Ink {} remaining level {}", thing.getUID().toString(), inkName, inkRemaining);
+
+                        switch (inkName.toLowerCase()) {
+                            case "black":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_BLACK_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "yellow":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_YELLOW_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "magenta":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_MAGENTA_LEVEL),
+                                        new DecimalType(inkRemaining));
+
+                                break;
+
+                            case "cyan":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_CYAN_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "cyanmagentayellow":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_COLOUR_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                        }
+                    }
+                } catch (Exception ex) {
```

Comment:
Is it possible to catch more specific exceptions (e.g. `IllegalArgumentException`)?